### PR TITLE
Adds two new traumas, Mind Echo and Existential Crisis

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds two new traumas in the trauma pool.
Mind Echo: a mild trauma that records what the patient hears and says, and occasionally replaces heard or spoken messages with one of the recorded ones.
Existential Crisis: a special trauma that causes spontaneous temporary non-existence, which works like a desynchronizer does.

## Why It's Good For The Game

More variety in traumas!

## Changelog
:cl:
add: Added two new traumas in the trauma pool: Mind Echo (mild) and Existential Crisis (special).
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
Existential Crisis is an amazing name for it though, that's sick.

**Comment by ATH1909** [comment]
anconfusedrock kind of has a point, tbh. While I think that mind echo is unique enough, is there anything that we can change about existential crisis to make it more unique?

**Comment by OverlyGeneric** [comment]
> anconfusedrock kind of has a point, tbh. While I think that mind echo is unique enough, is there anything that we can change about existential crisis to make it more unique?

Be severely depressed and slowly fade away the sadder you are.

**Comment by tralezab** [comment]
Having your mood change how "real" you are and a low mood making you not exist anymore would be very cool

**Review by coiax — CHANGES_REQUESTED** [review]

**Comment by Grazyn** [comment]
There is an actual mental illness in which you believe you don't exist, it's called Cotard's delusion.

**Comment by XDTM** [comment]
Now i'm torn between the wordplay name or the obscure based-in-reality one...
Although reading the wiki page it can lead to delusions of immortality, which can be a nice idea for a separate special trauma in some form

**Comment by ATH1909** [comment]
> Although reading the wiki page it can lead to delusions of immortality, which can be a nice idea for a separate special trauma in some form

We already have anosognosia, just so you know.

**Comment by Grazyn** [comment]
I admit, existential crisis is way funnier, also I advise against using the name cotard for a "immortality delusion" trauma because Cotard's is mostly known as the "delusion of death" thing in popular culture, despite the actual article saying that immortality is the majority

**Comment by ExcessiveUseOfCobblestone** [comment]
is existential crisis just the immortality talisman

**Comment by davethwave** [comment]
Thought of a fun feature, how about if you use the desynchronizer too long for perhaps 7 minutes total or something you gain the existential crisis trauma?

**Comment by XDTM** [comment]
If we do that it might be worth doing similar stuff for other traumas, like gaining phobias from certain events, as long as it's a rare event.
@ExcessiveUseOfCobblestone i guess, but it's completely incontrollable as well as invisible, and the patient can do stuff while not-existing, like when using a desynchronizer

**Comment by ExcessiveUseOfCobblestone** [comment]
I see...

File: code/datums/brain_damage/special.dm [hunk 1]
```diff
@@ -182,8 +182,56 @@
...
+	scan_desc = "existential crisis"
+	gain_text = "<span class='notice'>You feel less real.</span>"
+	lose_text = "<span class='warning'>You feel more substantial again.</span>"
+	var/obj/effect/abstract/sync_holder/veil/veil
+	var/next_crisis = 0
+
+/datum/brain_trauma/special/existential_crisis/on_life()
+	..()
+	if(!veil && world.time > next_crisis && prob(3))
+		fade_out()
```
[the remark this answers is not in the document]
Comment (ExcessiveUseOfCobblestone) [reply code/datums/brain_damage/special.dm]: This issue will need to be ratified

[the remark this answers is not in the document]
Comment (XDTM) [reply code/datums/brain_damage/special.dm]: i'll have some more time to code this weekend, i'll make sure to prevent those cases

**Comment by ExcessiveUseOfCobblestone** [comment]
Update?

**Comment by XDTM** [comment]
Made it so it only triggers when loc is a turf, should get rid of buggy interactions with containers altogether.

File: code/datums/brain_damage/mild.dm [hunk 2]
```diff
@@ -226,3 +226,39 @@
...
+	gain_text = "<span class='warning'>You feel a faint echo of your thoughts...</span>"
+	lose_text = "<span class='notice'>The faint echo fades away.</span>"
+	var/list/hear_dejavu = list()
+	var/list/speak_dejavu = list()
+
+/datum/brain_trauma/mild/mind_echo/on_hear(message, speaker, message_language, raw_message, radio_freq)
+	if(hear_dejavu.len >= 5)
+		if(prob(15))
+			var/deja_vu = pick_n_take(hear_dejavu)
+			var/new_message = replacetext(message, regex("\".+\"", "gi"),"\"[deja_vu]\"") //Quotes included to avoid cases where someone says part of their name
```
Comment (AnturK) [inline code/datums/brain_damage/mild.dm]: Should store this regex since it's static.

File: code/datums/brain_damage/special.dm [hunk 3]
```diff
@@ -182,8 +182,57 @@
...
+
+/datum/brain_trauma/special/existential_crisis/on_lose()
+	if(veil)
+		fade_in()
+	..()
+
+/datum/brain_trauma/special/existential_crisis/proc/fade_out()
+	if(veil)
+		return
+	var/duration = pick(50, 450)
```
Comment (AnturK) [inline code/datums/brain_damage/special.dm]: Was this supposed to be rand or it's correct ?

**Review by AnturK — CHANGES_REQUESTED** [review]
Just a nitpick

File: code/datums/brain_damage/special.dm (hunk 3 above)
Comment (XDTM) [reply code/datums/brain_damage/special.dm]: oops, yeah it's rand

**Comment by XDTM** [comment]
@optimumtact i found this old PR in my list; i fixed the conflicts and addressed the AnturK review, and it's functional. 

This PR can be reopened, or i could open another one from the branch if it's an issue.

**Comment by ExcessiveUseOfCobblestone** [comment]
Old PRs regarding balance/features should be re-PR'd (preferably with link to old) so it shows on the servers.

**Comment by XDTM** [comment]
Alright, will do

**Pull request closed without merging**